src/site/content/en/blog/google-search-sw/index.md

```diff
@@ -0,0 +1,463 @@
+least 2 gigabytes of RAM, and enough free storage space, then a
+[service worker will be registered](https://developers.google.com/web/fundamentals/primers/service-workers/lifecycle#the_first_service_worker).
+Browsers or devices that don't meet that criteria won't end up with a service
+worker, but they'll still see the same Google Search experience as they always
+have.
+
+One side benefit of this selective registration is that it has allowed the
+Search team to ship a smaller, more efficient service worker. Knowing that only
+fairly modern browsers will end up running the service worker code means
+eliminating the overhead of transpilation and polyfills needed to target
+browsers on which registration will never take place. This ended up cutting out
+around 8 kilobytes of uncompressed JavaScript code from the total size of the
+service worker's implementation.
+
+### Service worker scopes
+
+Once the Search team ran enough latency experiments and were confident that
+using navigation preload offered them a viable, latency-neutral path for using a
+service worker, some practical issues started moving to the forefront. One of
+those issues has to do with service worker's
+[scoping rules](https://developers.google.com/web/ilt/pwa/introduction-to-service-worker#registration_and_scope).
+A service worker's scope determines which pages it can potentially take control
+of, and scoping works based on URL path prefix. For domains that host a single
+web app, this isn't an issue, as you'd normally just use a service worker with
+the maximal scope of `/`, which could take control of any page under the domain.
+But Google Search's URL structure is a little more complicated.
+
+If the service worker were given the maximal scope of `/`, it would end up being
+able to take control of any page hosted under `www.google.com` (or the regional
+equivalent), and there are URLs under that domain that have nothing to do with
+Google Search. A more reasonable, restrictive scope would be `/search`, which at
+least would eliminate URLs completely unrelated to search results. But
+unfortunately, even that `/search` URL path is shared amongst different flavor
+of Google Search results, with URL query parameters determining which specific
+type of search result is shown. Some of those flavors use completely different
+codebases than the traditional web search result page—for example, Image Search
+and Shopping Search are both served under the `/search` URL path with different
+query parameters, but neither of those interfaces were ready to ship their own
+service worker experience (yet).
+
+**Solution**
+
+While there are [some proposals](https://github.com/w3c/ServiceWorker/issues/1373)
+that could allow for something more powerful than URL path prefixes to determine
+service worker scopes, in the present, the Google Search team was stuck
+deploying a service worker that effectively needed to do nothing for a subset of
```

Comment:
Small changes to lean this first sentence a bit:

While there are some proposals that allow for something more powerful than URL path prefixes to determine service worker scopes, the Google Search team was stuck deploying a service worker that did nothing for a subset of pages it controlled.